crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -33,6 +33,9 @@ class ApplicationController < ActionController::Base
     Crowbar::Installer.successful? || \
     Rails.env.test?
   }
+  before_filter :upgrade, if: proc {
+    File.exist?("/var/lib/crowbar/upgrade/progress.yml")
```

Comment:
Don't know. IIRC you were the one to consider this as needed.
1. How would the API respond if the file is missing?
2. When will this file be deleted and by which component? (or maybe this is done already?)
3. When does "upgrade" begin? (i.e. when is this file created?)
4. What about "cancel"? Does it remove the file?